cmd/srcd/cmd/init.go

```diff
@@ -28,39 +28,35 @@ import (
 var initCmd = &cobra.Command{
 	Use:   "init",
 	Short: "Starts the daemon or restarts it if already running.",
+	Args:  cobra.MaximumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		if len(args) > 1 {
-			logrus.Fatal("invalid number of arguments given, expecting 0 or 1")
-		}
-
-		err := daemon.Kill()
-		if err != nil {
-			logrus.Fatal(err)
-		}
-
+		var err error
 		var workdir string
+
 		if len(args) > 0 {
 			workdir = args[0]
 		}
 
 		workdir = strings.TrimSpace(workdir)
 		if workdir == "" {
 			workdir, err = os.Getwd()
-			if err != nil {
-				logrus.Fatal(err)
-			}
 		} else {
 			workdir, err = filepath.Abs(workdir)
-			if err != nil {
-				logrus.Fatal(err)
-			}
+		}
+		if err != nil {
+			logrus.Fatal(err)
 		}
 
 		info, err := os.Stat(workdir)
 		if err != nil || !info.IsDir() {
 			logrus.Fatalf("path %q is not a valid working directory", workdir)
 		}
```

Comment:
Actually I think that everything until this line should go inside a custom function provided to the `Args` key given that it still contains validation logic. But it would require recomputing the `workdir` variable. One way to avoid this is to define `var workdir string` at top-level, and it will be initialized during the args validation.

Reply:
What you mention might be cleaner if we had to reuse the same validation. But the current code is small, maybe it's not worth it to have a package variable just for that.

Reply:
yup exactly :)